<!-- Launch file for the rrt-detector and the assigner -->


<launch>
    <arg name="eta" value="1.2"/>
    <arg name="Geta" value="1.2"/>
    <arg name="info_radius" value="1.0"/>
    <arg name="info_multiplier" value="4.0"/>
    <arg name="hysteresis_radius" value="3.0"/>
    <arg name="hysteresis_gain" value="3.0"/>
    <arg name="costmap_clearing_threshold" value="60"/>
    <arg name="time_per_meter" value="6.0"/>
    <arg name="delay_after_assignment" value="0.3"/>   
    <arg name="bandwith_cluster" value="0.32"/>   
    <arg name="frequency1" value="50.0"/>
    <arg name="frequency2" value="20.0"/>
    <arg name="detected_frontier" value="/detected_points"/>
    <arg name="filtered_frontier" value="/filtered_points"/>
    <arg name="rd_metric" value="18.0"/>
    <!-- robot name space definition -->
    <arg name="first_tb3"  default="tb3_1"/>
    <arg name="second_tb3" default="tb3_2"/>
    <arg name="third_tb3"  default="tb3_0"/>

    <arg name="global_map" value ="/map"/>
    <arg name="global_frame" value ="map"/>
    <arg name="global_costmap_topic" value="/move_base_node/global_costmap/costmap"/>
    <arg name="plan_service" value="/move_base_node/NavfnROS/make_plan"/>
    <arg name="robot_base_link" value="base_link"/>
    <arg name="move_base_service" value="/move_base"/>

    <!-- for starting the simulation in the initial poitn of the data -->
    <!-- <arg name="initialStartPts" value ="0.39,0.58"/> -->
    <!-- <arg name="initialStartPts" value ="1.98,2.43"/> -->
    <arg name="odom_topic_startpts" value="/tb3_2/odom"/>
    <arg name="robot_frame" value="/tb3_2/base_link"/>
    
  
  
    <node pkg="tmrrt_exploration" type="boundary.py" name="exploration_boundary" output="screen">
      <param name="mapFrame" value="world"/>
      <param name="n_point" value="4"/>
      <param name="topicInput" value="clicked_point"/>
      <param name="topicOutput" value="exploration_boundary"/>
      <param name="frequency" value="0.5"/>
      <param name="mapTopic" value="/map"/>
      <!-- <param name="initialStartPts" value="$(arg initialStartPts)"/> -->
      <param name="odom_topic_startpts" value="$(arg odom_topic_startpts)"/>
      <param name="robot_frame" value="$(arg robot_frame)"/>
      </node>
      
    <node pkg="tmrrt_exploration" type="global_rrt_detector" name="global_rrt_detector" output="screen">
      <param name="eta" value="$(arg Geta)"/>
      <param name="map_topic" value="/map"/>
    </node>
  
    <node pkg="tmrrt_exploration" type="local_rrt_detector" name="$(arg first_tb3)_rrt_detector" output="screen">
      <param name="eta" value="$(arg eta)"/>
      <param name="map_topic" value="/$(arg first_tb3)/map"/>
      <param name="robot_frame" value="/$(arg first_tb3)/base_link"/>
    </node>
  
    <node pkg="tmrrt_exploration" type="local_rrt_detector" name="$(arg second_tb3)_rrt_detector" output="screen">
      <param name="eta" value="$(arg eta)"/>
      <param name="map_topic" value="/$(arg second_tb3)/map"/>
      <param name="robot_frame" value="/$(arg second_tb3)/base_link"/>
    </node>
  
    <node pkg="tmrrt_exploration" type="local_rrt_detector" name="$(arg third_tb3)_rrt_detector" output="screen">
      <param name="eta" value="$(arg eta)"/>
      <param name="map_topic" value="/$(arg third_tb3)/map"/>
      <param name="robot_frame" value="/$(arg third_tb3)/base_link"/>
    </node>
    

    <node pkg="tmrrt_exploration" type="filter.py" name="filter" output="screen">
      <param name="map_topic" value="$(arg global_map)"/>
      <param name="info_radius" value="$(arg info_radius)"/> 
      <param name="costmap_clearing_threshold" value="$(arg costmap_clearing_threshold)"/> 
      <param name="global_costmap_topic" value="$(arg global_costmap_topic)"/> 
      <param name="goals_topic" value="$(arg detected_frontier)"/> 
      <param name="robot_namelist" value="$(arg first_tb3),$(arg second_tb3),$(arg third_tb3)"/>
      <param name="rate" value="$(arg frequency1)"/>
      <param name="local_map" value="/map"/>
      <param name="bandwith_cluster" value="$(arg bandwith_cluster)"/>
    </node>

    <node pkg="tmrrt_exploration" type="assigner.py" name="assigner" output="screen">
      <param name="map_topic" value="/map"/>
      <param name="global_frame" value="map"/>
      <param name="info_radius" value="$(arg info_radius)"/> 
      <param name="info_multiplier" value="$(arg info_multiplier)"/> 
      <param name="hysteresis_radius" value="$(arg hysteresis_radius)"/> 
      <param name="hysteresis_gain" value="$(arg hysteresis_gain)"/> 
      <param name="frontiers_topic" value="/filtered_points"/> 
      <param name="robot_namelist" value="$(arg first_tb3),$(arg second_tb3),$(arg third_tb3)"/>
      <param name="time_per_meter" value="$(arg time_per_meter)"/>
      <param name="delay_after_assignment" value="$(arg delay_after_assignment)"/>
      <param name="rate" value="$(arg frequency2)"/>
      <param name="plan_service" value="$(arg plan_service)"/>
      <param name="base_link" value="$(arg robot_base_link)"/>
      <param name="move_base_service" value="$(arg move_base_service)"/>
      <param name="rd_metric" value="$(arg rd_metric)"/>
    </node>
  </launch>
